src/users/views.py

```diff
@@ -228,26 +228,23 @@ class MentorsSearchView(generics.ListAPIView):
     def filter_queryset(self, queryset):
         queryset = queryset.exclude(profile__user=self.request.user) 
 
-        major = 'all'
-        year = 'all'
-        
-        if 'major' in self.request.GET:
-            major = self.request.GET['major']
-        if 'year' in self.request.GET:
-            year = self.request.GET['year']
-
-        q = Q()
-        if major != 'all':
-            q &= Q(major__name=major)
-        if year != 'all':
-            q &= Q(profile__year=year)
-
-        queryset = queryset.filter(q)
-        
+
+        if 'query' in self.request.GET:
+            query = self.request.GET['query']
+            query = query.split(' ')
+
         
+            for item in query:
+                queryset = queryset.filter(
+                    Q(major__name__icontains = item) | 
+                    Q(profile__year__icontains = item) | 
+                    Q(profile__user__first_name__icontains = item) |
+                    Q(profile__user__last_name__icontains = item)
```

Comment:
can we also translate `'first'` to `'1st'` and so on?  i'm a little worried about the odds of a user searching precisely "1st"

Reply:
As in if the original query contains 'first', add '1st' to the list of items iterated over?